app/models/host_status/global.rb

```diff
@@ -0,0 +1,44 @@
+module HostStatus
+  class Global
+    OK = 0
+    WARN = 1
+    ERROR = 2
+
+    attr_accessor :status
+
+    def self.build(statuses, options = {})
+      if options.has_key?(:last_reports) && (configuration = statuses.find { |s| s.type == 'HostStatus::ConfigurationStatus' })
+        cached_report = options[:last_reports].find { |r| r.host_id == configuration.host_id }
+        configuration.last_report = cached_report
```

Comment:
Could we just pass all options into a status, e.g. into to_global as an argument or another initialisation sort of method?  This may come up again if new statuses are added that require looking at other DB records.